### PR TITLE
Nlog Implementation 

### PR DESCRIPTION
**Pull request by manic1990**
Implemented Nlog
Added unit test case for API controller Get Booking method with ILogger mocking

### CODE REVIEW

File: BookingSystem.DAL/Migrations/20211222102045_IntialMigration.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace BookingSystem.DAL.Migrations
+{
+    public partial class IntialMigration : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
```
Comment (vivekr2021) [inline BookingSystem.DAL/Migrations/20211222102045_IntialMigration.cs:7]: Why do we have the empty Migration Class? Can you please remove it instead?

File: BookingSystem/appsettings.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
     }
   },
   "ConnectionStrings": {
-    "DbConnection": "Server=localhost;User ID=postgres;Password=1234;Database=BookingSystem;Integrated Security=true;Pooling=true;"
+    "DbConnection": "Server=localhost;User ID=postgres;Password=123@T123;Database=BookingSystem;Integrated Security=true;Pooling=true;"
```
Comment (vivekr2021) [inline BookingSystem/appsettings.json:18]: Don't need to commit the appsettings file if there is no reasonable change. You can undo it before you commit the code

**Review by vivekr2021 — CHANGES_REQUESTED** [review]

File: BookingSystem/appsettings.json (hunk 2 above)
Comment (manic1990) [reply BookingSystem/appsettings.json:18]: Updated common password

File: BookingSystem.DAL/Migrations/20211222102045_IntialMigration.cs (hunk 1 above)
Comment (manic1990) [reply BookingSystem.DAL/Migrations/20211222102045_IntialMigration.cs]: Removed empty migration class

**Pull request merged**